pkg/flare/flare.go

```diff
@@ -31,6 +31,29 @@ type flareResponse struct {
 
 // SendFlareWithHostname sends a flare with a set hostname
 func SendFlareWithHostname(archivePath string, caseID string, email string, hostname string) (string, error) {
+	r, err := readAndPostFlareFileWithRedirects(archivePath, caseID, email, hostname)
+	return analyzeResponse(r, err)
+}
+
+func readAndPostFlareFileWithRedirects(archivePath string, caseID string, email string, hostname string) (*http.Response, error) {
+	var url = mkURL(caseID)
+	var redirectHops = 0
+
+	// Handle redirects manually. Go's http.Client doesn't know how to do it when it can't seek
+	// back to the beginning of the body. Since we are using a pipe, seeking isn't possible.
+	// Re-sending a POST is only legal for status 307, so we only need to check for that code.
+	for {
```

Comment:
Sorry to sound pick, but I find clearer to have the redirectHops count directly in the for call. Knowing when this for would directly by looking at it.

Reply:
👍